command/format/diff.go

```diff
@@ -1104,7 +1117,7 @@ func (p *blockBodyDiffPrinter) writeValueDiff(old, new cty.Value, indent int, pa
 					action = plans.Create
 				} else if new.HasIndex(kV).False() {
 					action = plans.Delete
-				} else if eqV := old.Index(kV).Equals(new.Index(kV)); eqV.IsKnown() && eqV.True() {
+				} else if eqV := unmarkedOld.Index(kV).Equals(unmarkedNew.Index(kV)); eqV.IsKnown() && eqV.True() {
```

Comment:
Huh. I think the case you're talking about can arise, but I'm also a little stuck on the "what we should do here". Would it be fair to say I'm considering this part of the larger "what should we do when sensitivity changes" question? (which is as yet undecided).

Reply:
I was thinking it's more useful to display the change, but perhaps (and it would mean less complex diff work) here, there is an option similar to the previous, but a bit more information:

We could cover cases: Both values sensitive, old is sensitive + new is not, old is not sensitive and new sensitive with messages saying _that_ ex. (sensitive) => (sensitive), (non-sensitive value) => (sensitive), (sensitive) => (non-sensitive value).  Wording flexible of course, but I think that approach might _actually_ give the most information and foot-gun-avoidance you're describing here.

Reply:
I like this compromise of displaying something like `(non-sensitive value)`, so an attentive reader can get an indication that the value is _about_ to become visible but giving them one more chance to change their minds about it before the value gets disclosed.

I'm a little unsure about the exact terminology to use here. I worry that `(non-sensitive value)` might not be sufficiently clear for a reader to understand the implications about it. Part of me is wondering if "sensitive to non-sensitive" is a significant enough situation to warrant a "louder" warning, perhaps like this:

```
  # Warning: this attribute value will no longer be marked as sensitive
  # after applying this change.
  ~ foo = (sensitive)
```

(I guess, with our typical yellow-text treatment for the "Warning: " prefix, which hopefully we've trained color-sighted users to notice when scanning over the output.)

The transition from non-sensitive to sensitive seems like it has a similar concern but without the danger level that would justify a "loud" presentation, so I'd be less worried about the clarity of `(non-sensitive value)` in that case where the stakes are lower:

```
  ~ foo = (non-sensitive value) => (sensitive value)
```